### ISSUE TITLE
Implement logic so that or operators can be passed in query parameters of a request

### ISSUE THREAD

**Issue by vschaefer** [lucascranach/cranach-api#70]

**Comment by vschaefer** [comment]
Hello, ich habe mir mal ein paar Gedanken zu den Oder-Verknüpfungen gemacht. Und das ist doch gar nicht so so trivial diese in die URL zur Abfrage der API einzubauen.
Am nächsten liegen würde ja etwas in der Art:
`localhost:8080/?entity_type:eq=PAINTING||dating_begin:eq=1920&dating_end:eq=194`
Hierbei würde die Oder-Verknüpfung durch ||  und die Und-Verknüpfungen durch &  realisiert. Wobei die Oder-Verknüpfungen Vorrang vor den Und-Verknüpfungen haben.
Mit gesetzten Klammern würde das dann wie folgt aussehen:
`localhost:8080/?(entity_type:eq=PAINTING||dating_begin:eq=1920)&dating_end:eq=1940`
Das Problem dabei ist, dass ein Query Parameter so definiert ist, dass vor dem Gleichheitszeichen immer der Attributname steht, und danach der Attributwert. Und in diesem recht simplen Fall geht es schon nicht auf.
`entity_type:eq=` ist der Attributname bzw. Filtername + Operator und `PAINTING||dating_begin:eq=1920` der Attributwert bzw. Wert nach dem gefiltert wird. Aber das wäre ja nicht korrekt, da hinter `PAINTING` noch ein weiter Filter steht, der mit ODER verknüpft ist. Das heißt diese Variante würde nicht funktionieren.
Mein Vorschlag wäre in einem einzigen Query Parameter die gesamten Filter zu definieren und für die Zeichen, welche zur Definition von Query Parametern reserviert sind auszutauschen. Das wären `=` und `&`.
Ich könnte mir innerhalb des Filters dann folgende Operatoren vorstellen
UND = `++`
ODER = `||`
= = `::`
Die obere Abfrage würde dann wie folgt aussehen:
`localhost:8080/?filter=entity_type.eq::PAINTING||dating_begin.eq::1920++dating_end.eq::1940`
Was meint ihr? Ideen und Vorschläge sind willkommen :slightly_smiling_face:

**Comment by vschaefer** [comment]
Wenn der Datierungs-Filter gesetzt wird, wird eine Elasticsearch Abfrage zur Selektierung der Daten abgesetzt, welche bisher durch Übergabe von Queryparametern an die API nicht abbildbar ist. Deshalb ist die Filterlogik für den Datierungs-Filter in der API festgebacken. Dies macht die API ein Stück weit starrer und das Verhalten undurchsichtiger. Um dieses Manko zu beseitigen soll die Abfragelogik der API erweitert werden, so dass auch einzelne Filterparamter mit einem  OR-Operator verknüpft werden können.
Zur Veranschaulichung:

Wenn der Datierungsfilter gesetzt wird, sieht der Request an die API wie folgt aus
https://mivs02.gm.fh-koeln.de/?dating_begin:gte=1490&dating_end:lte=1588

Übersetzt: Zeige alle Werke mit datin_begin >= 1490 UND dating_end <= 1588.
Dies ist aber nicht identisch zu dem, wie die API die Daten filtert, bzw. wie die Abfrage ans Elasticsearch aussieht.
Diese sieht wie folgt aus:
dating_begin >= 1490 OR dating_end <= 1490) UND (dating_end <= 1588 OR dating_begin >= 1588)

**Comment by jhfpereira** [comment]
Da `&&` eine höhere Reihenfolge hat, müsste `dating_end.lte::1490&&dating_end.lte::1588` wahr sein.
Weshalb wohl schon mit Klammern gearbeitet werden muss:
`https://mivs02.gm.fh-koeln.de/?filter=(dating_begin.gte::1490||dating_end.lte::1490)&&(dating_end.lte::1588||dating_begin.gte::1588)`

**Comment by vschaefer** [comment]
Stimmt, das macht die Sache komplexer. Oder wir ändern die Rangfolge, was aber die allgemein Regeln brechen würde.

**Comment by vschaefer** [comment]
Dieses Feature wird auf den Backlog gelegt, da bisher nur ein Anwendungsfall existiert, welches das Feature nutzt und der Entwicklungsaufwand dafür hoch ist.
